internal/martian/proxy_handler.go

```diff
@@ -337,27 +337,36 @@ func (p proxyHandler) handleRequest(rw http.ResponseWriter, req *http.Request) {
 	}
 }
 
-type writeFlusher struct {
-	rw io.Writer
-	rc *http.ResponseController
+type h2Writer struct {
+	w     io.Writer
+	flush func() error
+	close func() error
 }
 
-func (w writeFlusher) Write(p []byte) (n int, err error) {
-	n, err = w.rw.Write(p)
+func makeH2Writer(rw http.ResponseWriter, rc *http.ResponseController, req *http.Request) *h2Writer {
```

Comment:
Return a pointer? Previous flusher was a copy, and the methods receive a copy too, so I'm double checking if that's on purpose.

Reply:
Nah, that's too fast generation...

Reply:
Fixed